### PR TITLE
Test in-cluster console for e2e tests

### PR DESCRIPTION
**Pull request by spadgett**
Previously we were running the bridge binary locally. Instead test the in-cluster console, which tests the built console image.

/cc @jcaianirh @alecmerdler @benjaminapetersen 

https://jira.coreos.com/browse/CONSOLE-1267
https://jira.coreos.com/browse/CONSOLE-1266

### CODE REVIEW

**Comment by benjaminapetersen** [comment]
```
level=info msg="Waiting up to 30m0s for the cluster to initialize..."
level=fatal msg="failed to initialize the cluster: Cluster operator network has not yet reported success"
```

**Comment by spadgett** [comment]
/retest

```
level=info msg="Creating cluster..."
level=error
level=error msg="Error: Error applying plan:"
level=error
level=error msg="1 error occurred:"
level=error msg="\t* module.vpc.aws_route_table_association.route_net[3]: 1 error occurred:"
level=error msg="\t* aws_route_table_association.route_net.3: timeout while waiting for state to become 'success' (timeout: 5m0s)"
```

**Comment by spadgett** [comment]
```
+ ./test-gui.sh e2e
+ copyArtifacts
++ pwd
+ echo 'Copying artifacts from /go/src/github.com/openshift/console...'
Copying artifacts from /go/src/github.com/openshift/console...
+ cp -rv ./frontend/gui_test_screenshots /tmp/artifacts/gui_test_screenshots
cp: cannot stat './frontend/gui_test_screenshots': No such file or directory
```

Not sure what failed in this run. It looks like the e2e tests didn't start.

I added a guard in `copyArtifacts` to avoid that error and added `set -x` to `test-gui.sh` for better logging.

**Comment by spadgett** [comment]
> level=fatal msg="failed to initialize the cluster: Cluster operator network has not yet reported success"

/retest

**Comment by spadgett** [comment]
Looks like the call to `yarn run webdriver-update` is failing. We send output to `/dev/null` so it's not clear why.

**Comment by spadgett** [comment]
Ah, I forgot to add `yarn install` since we removed `build-frontend.sh`

File: frontend/integration-tests/protractor.conf.ts [hunk 1]
```diff
@@ -67,6 +67,9 @@ export const config: Config = {
       consoleLogStream.write(`${format.apply(null, [`[${level.name}]`, messageStr])}\n`);
     });
 
+    const url = await browser.getCurrentUrl();
+    console.log('Last browser URL', url);
```
Comment (benjaminapetersen) [inline frontend/integration-tests/protractor.conf.ts]: +1

**Comment by spadgett** [comment]
Closing and reopening to kick the tests

**Comment by spadgett** [comment]
OK we had one login succeed with a different e2e flake and the last failure got to the login redirect.

> Last browser URL https://console-openshift-console.apps.ci-op-psn4k3pg-514ba.origin-ci-int-aws.dev.rhcloud.com/auth/login

/retest

**Comment by spadgett** [comment]
Login failed due to bug 1679272. We might need to fold #1206 and this into one PR.

> Last browser URL https://console-openshift-console.apps.ci-op-psn4k3pg-514ba.origin-ci-int-aws.dev.rhcloud.com/error?error=invalid_code&error_type=auth

/retest

**Comment by spadgett** [comment]
/retest

```
Failed Specs:

1) Performance test : downloads new bundle for operator-management
   Error: Failed expectation
       at Object.<anonymous> /go/src/github.com/openshift/console/frontend/integration-tests/tests/performance.scenario.ts:96:30
       at step /go/src/github.com/openshift/console/frontend/integration-tests/tests/performance.scenario.ts:32:23
       at Object.next /go/src/github.com/openshift/console/frontend/integration-tests/tests/performance.scenario.ts:13:53
       at fulfilled /go/src/github.com/openshift/console/frontend/integration-tests/tests/performance.scenario.ts:4:58
       at process._tickCallback internal/process/next_tick.js:68:7
   TypeError: Cannot read property 'decodedBodySize' of null
       at Object.<anonymous> /go/src/github.com/openshift/console/frontend/integration-tests/tests/performance.scenario.ts:101:36
       at step /go/src/github.com/openshift/console/frontend/integration-tests/tests/performance.scenario.ts:32:23
       at Object.next /go/src/github.com/openshift/console/frontend/integration-tests/tests/performance.scenario.ts:13:53
       at fulfilled /go/src/github.com/openshift/console/frontend/integration-tests/tests/performance.scenario.ts:4:58
       at process._tickCallback internal/process/next_tick.js:68:7
   From: Task: Run it"downloads new bundle for operator-management" in control flow
       at UserContext.<anonymous> /go/src/github.com/openshift/console/frontend/node_modules/jasminewd2/index.js:94:19
   From asynchronous test: 
   Error
       at /go/src/github.com/openshift/console/frontend/integration-tests/tests/performance.scenario.ts:86:5
       at /go/src/github.com/openshift/console/frontend/node_modules/immutable/dist/immutable.js:2699:43
       at List.__iterate /go/src/github.com/openshift/console/frontend/node_modules/immutable/dist/immutable.js:2206:13
       at OrderedMap.__iterate /go/src/github.com/openshift/console/frontend/node_modules/immutable/dist/immutable.js:2698:25
       at OrderedMap.forEach /go/src/github.com/openshift/console/frontend/node_modules/immutable/dist/immutable.js:4381:19
       at Suite.<anonymous> /go/src/github.com/openshift/console/frontend/integration-tests/tests/performance.scenario.ts:78:17
       at Env.<anonymous> /go/src/github.com/openshift/console/frontend/node_modules/jasmine-fail-fast/dist/jasmine-fail-fast.js:57:26
```

**Comment by spadgett** [comment]
> level=fatal msg="failed to initialize the cluster: Cluster operator network has not yet reported success"

/retest

**Comment by spadgett** [comment]
Failed because of bug 1679272 (needs #1206)

/retest

**Comment by spadgett** [comment]
Failed again because of 1679272

/retest

**Comment by spadgett** [comment]
I opened #1207 which combines the two fixes to unblock the queue

File: frontend/integration-tests/protractor.conf.ts [hunk 2]
```diff
@@ -31,6 +31,7 @@ export const config: Config = {
   plugins: [failFast.init()],
   capabilities: {
     browserName: 'chrome',
+    acceptInsecureCerts : true,
```
Comment (enj) [inline frontend/integration-tests/protractor.conf.ts]: What certs does this use?  Could it be having the same SA crt changes under it issues?

File: test-prow-e2e.sh [hunk 3]
```diff
@@ -2,23 +2,27 @@
...
+    echo "Copying artifacts from $(pwd)..."
+    cp -rv "$SCREENSHOTS_DIR" "${ARTIFACT_DIR}/gui_test_screenshots"
+  fi
 }
 
 trap copyArtifacts EXIT
 
-./build.sh
+export BRIDGE_AUTH_USERNAME=kubeadmin
+# don't log kubeadmin-password
```
Comment (enj) [inline test-prow-e2e.sh]: :smile:

File: test-prow-e2e.sh [hunk 4]
```diff
@@ -2,23 +2,27 @@
...
+set +x
+export BRIDGE_AUTH_PASSWORD="$(cat "${INSTALLER_DIR}/auth/kubeadmin-password")"
+set -x
+export BRIDGE_BASE_ADDRESS="https://$(oc get route console -n openshift-console -o jsonpath='{.spec.host}')"
```
Comment (enj) [inline test-prow-e2e.sh]: At some point you should be able to get this from your top level config

**Comment by spadgett** [comment]
Same performance test failure. Something is different when running in cluster:

```
 Failed Specs:

1) Performance test : downloads new bundle for operator-management
   Error: Failed expectation
       at Object.<anonymous> /go/src/github.com/openshift/console/frontend/integration-tests/tests/performance.scenario.ts:96:30
       at step /go/src/github.com/openshift/console/frontend/integration-tests/tests/performance.scenario.ts:32:23
       at Object.next /go/src/github.com/openshift/console/frontend/integration-tests/tests/performance.scenario.ts:13:53
       at fulfilled /go/src/github.com/openshift/console/frontend/integration-tests/tests/performance.scenario.ts:4:58
       at process._tickCallback internal/process/next_tick.js:68:7
   TypeError: Cannot read property 'decodedBodySize' of null
       at Object.<anonymous> /go/src/github.com/openshift/console/frontend/integration-tests/tests/performance.scenario.ts:101:36
       at step /go/src/github.com/openshift/console/frontend/integration-tests/tests/performance.scenario.ts:32:23
       at Object.next /go/src/github.com/openshift/console/frontend/integration-tests/tests/performance.scenario.ts:13:53
       at fulfilled /go/src/github.com/openshift/console/frontend/integration-tests/tests/performance.scenario.ts:4:58
       at process._tickCallback internal/process/next_tick.js:68:7
   From: Task: Run it"downloads new bundle for operator-management" in control flow
       at UserContext.<anonymous> /go/src/github.com/openshift/console/frontend/node_modules/jasminewd2/index.js:94:19
   From asynchronous test: 
   Error
       at /go/src/github.com/openshift/console/frontend/integration-tests/tests/performance.scenario.ts:86:5
       at /go/src/github.com/openshift/console/frontend/node_modules/immutable/dist/immutable.js:2699:43
       at List.__iterate /go/src/github.com/openshift/console/frontend/node_modules/immutable/dist/immutable.js:2206:13
       at OrderedMap.__iterate /go/src/github.com/openshift/console/frontend/node_modules/immutable/dist/immutable.js:2698:25
       at OrderedMap.forEach /go/src/github.com/openshift/console/frontend/node_modules/immutable/dist/immutable.js:4381:19
       at Suite.<anonymous> /go/src/github.com/openshift/console/frontend/integration-tests/tests/performance.scenario.ts:78:17
       at Env.<anonymous> /go/src/github.com/openshift/console/frontend/node_modules/jasmine-fail-fast/dist/jasmine-fail-fast.js:57:26
```

File: frontend/integration-tests/protractor.conf.ts (hunk 2 above)
Comment (spadgett) [reply frontend/integration-tests/protractor.conf.ts]: This is to bypass the browser warning for self-signed certificates.

**Comment by spadgett** [comment]
Got all the way to the etcd operator tests

```
Interacting with the etcd OCS creates etcd Operator `Deployment` 1m41s
Error: Timeout - Async callback was not invoked within timeout specified by jasmine.DEFAULT_TIMEOUT_INTERVAL.
```

/retest

**Comment by spadgett** [comment]
@kyoto Thanks for helping me retest this 👍

**Comment by spadgett** [comment]
@alecmerdler OLM etcd scenario has failed 3 times. Can you help? Wondering if there's a real bug here:

https://openshift-gce-devel.appspot.com/build/origin-ci-test/pr-logs/pull/openshift_console/1197/pull-ci-openshift-console-master-console-e2e/673/

https://openshift-gce-devel.appspot.com/build/origin-ci-test/pr-logs/pull/openshift_console/1197/pull-ci-openshift-console-master-console-e2e/675/

https://openshift-gce-devel.appspot.com/build/origin-ci-test/pr-logs/pull/openshift_console/1197/pull-ci-openshift-console-master-console-e2e/678/

I don't see the etcd deployment in the screenshots:

https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_console/1197/pull-ci-openshift-console-master-console-e2e/673/artifacts/console-e2e/gui_test_screenshots/463125ea270e98ddf6d549a19bd5c35f.png

Pod logs should be in the artifacts if we need them. I might try to increase the timeout in case we're just not waiting long enough.

**Comment by spadgett** [comment]
Rate limited

/retest

```
level=fatal msg="failed to fetch Terraform Variables: failed to fetch dependency of \"Terraform Variables\": failed to fetch dependency of \"Bootstrap Ignition Config\": failed to fetch dependency of \"Common Manifests\": failed to generate asset \"DNS Config\": getting public zone for \"origin-ci-int-aws.dev.rhcloud.com\": listing hosted zones: Throttling: Rate exceeded\n\tstatus code: 400, request id: 329198a7-36af-11e9-8cbf-abfcb2fdb5d0"
```

**Comment by spadgett** [comment]
backend/frontend failures were infrastructure flakes (test-bin pod was evicted)

**Pull request merged**